testing/envtest/with.go

```diff
@@ -0,0 +1,26 @@
+package envtest
+
+import "os"
+
+func With(vars map[string]string, callback func() error) error {
+	defaults := make(map[string]string)
+
+	for k, v := range vars {
+		if val := os.Getenv(k); val != "" {
+			defaults[k] = val
+		}
+		os.Setenv(k, v)
+	}
+
+	err := callback()
+
+	for k := range vars {
+		os.Unsetenv(k)
+	}
+
+	for k, v := range defaults {
+		os.Setenv(k, v)
+	}
+
+	return err
```

Comment:
Why not return the callback () method directly?

Reply:
We have to restore the environment variables before returning. Sure we could use a defer, but it would be slower.